File: 643-MaximumAverageSubarrayI/643-MaximumAverageSubarrayI.cpp
// Last updated: 7/16/2026, 4:37:55 PM
class Solution {
public:
    double findMaxAverage(vector<int>& nums, int k) {
        int n = nums.size();
        int i = 0;
        int j = 0;
        double sum = 0;
        double max_sum =numeric_limits<double>::lowest() ;
        vector<int> arr;

        while (j < n) {
            sum += nums[j];
            if (nums[j]<n){
                arr.push_back(nums[j]);
            }
            if (j - i + 1 < k) {
                j++;
            } else if (j - i + 1 == k) {
                max_sum = max(max_sum, sum);
                sum -= nums[i];
                i++;
                j++;
            }
        }
        return max_sum/k;
    }
};

File: 643-MaximumAverageSubarrayI/643-MaximumAverageSubarrayI.py
# Last updated: 9/23/2026, 12:49:16 AM
1class Solution:
2    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
3        intervals.sort()  # Sort by start time
4        merged = []
5        prev = intervals[0]
6
7        for i in range(1, len(intervals)):
8            if intervals[i][0] <= prev[1]:  
9                prev[1] = max(prev[1], intervals[i][1])  
10            else:
11                merged.append(prev)
12                prev = intervals[i]
13
14        merged.append(prev)
15        return merged